docs/release/role-handbooks/release-lead/README.md

```diff
@@ -174,7 +174,7 @@ The goal of this task is to ensure we are always using the latest Go version for
 
 1. Keep track of new Go versions
 2. Bump the Go version in supported branches if necessary
-   <br>Prior art: [Bump go to v1.23.7](https://github.com/kubernetes-sigs/cluster-api/pull/11981)
+   <br>Prior art: [Bump go to v1.24.7](https://github.com/kubernetes-sigs/cluster-api/pull/11981)
```

Comment:
This doesn't match the PR anymore now. Let's revert the change to this file please